napari/layers/base/base.py

```diff
@@ -698,12 +700,14 @@ def _get_extent_world(self, data_extent):
     @property
     def extent(self) -> Extent:
         """Extent of layer in data and world coordinates."""
-        data = self._extent_data
-        return Extent(
-            data=data,
-            world=self._get_extent_world(data),
-            step=abs(self._data_to_world.scale),
-        )
+        extent_data = self._extent_data
+        if self._extent_cache is None:
+            self._extent_cache = Extent(
+                data=extent_data,
+                world=self._get_extent_world(extent_data),
+                step=abs(self.scale),
```

Comment:
This represents a change: `self.scale` (which you could maybe think of as pixel spacing) is not always equal to `self._data_to_world.scale` (which represents the scale of two composed transforms).

If I understand `extent` correctly (that's not a given yet), then I think you want `self._data_to_world.scale`.